File: source/server/src/main/java/com/profteam/bookgarden/dto/BookDto.java
package com.profteam.bookgarden.dto;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class BookDto {

    private String id;

    private String title;

    private String description;
    
    private String introduce;

    private int pageNum;

    private int amount;

    private Double price;
    
    private String image;

    private CategoryDto category;

    private AuthorDto author;

    private PublisherDto publisher;

}

File: source/client/customer/src/app/dto/response/book-dto.ts
export class BookDto {
    id: string;
    title: string;
    description: string;
    introduce: string;
    price: number;
    image: string;
    category: Category = new Category();
    author: Author = new Author();
    publisher: Publisher = new Publisher();
}

class Category {
    id: string;
    categoryTitle: string;
}

class Author {
    id: number;
    fullname: string;
}

class Publisher {
    id: number;
    name: string;
}
